### ISSUE TITLE
Template Haskell error: h$runInteractiveProcess is not defined

### ISSUE THREAD

**Issue by fgaray** [ghcjs/ghcjs#365]
Hello:

I am trying to build a QuasiQuoter to include jsx code in Haskell.

I have:

~~~{.haskell}
f = do
    return [jsx|
        <div>Hello</div>
        |]
~~~

And a simple QuasiQuoter that writes the output of "jsx" to a file with the help of the "process" library:

~~~{.haskell}
import Language.Haskell.TH
import Language.Haskell.TH.Quote
import System.Process
import GHC.IO.Handle

jsx :: QuasiQuoter
jsx = QuasiQuoter { quoteExp = exp }
    where
        exp :: String -> Q Exp
        exp str = do
            program <- runIO $ readCreateProcess (proc "jsx" []) str
            runIO $ writeFile "/tmp/debug" program
            [| () |]
~~~

This, however, fails with this error:

~~~
3 of 4] Compiling ReactJS          ( ReactJS.hs, ReactJS.js_o )
Linking template haskell (JQuery,JSX,ThRunner1)
uncaught exception in Haskell main thread: ReferenceError: h$runInteractiveProcess is not defined
ReferenceError: h$runInteractiveProcess is not defined
    at h$$processzm1zi2zi3zi0ZCSystemziProcessziInternalszizdj1_e_TJcgLH (eval at h$loadCodeStr (/home/felipe/.ghcjs/x86_64-linux-0.1.0-7.8.3/ghcjs/thrunner.js:139:10), <anonymous>:6860:11)
    at h$mainLoop (eval at h$loadCodeStr (/home/felipe/.ghcjs/x86_64-linux-0.1.0-7.8.3/ghcjs/thrunner.js:139:10), <anonymous>:9526:25)
    at eval (eval at h$loadCodeStr (/home/felipe/.ghcjs/x86_64-linux-0.1.0-7.8.3/ghcjs/thrunner.js:139:10), <anonymous>:9942:13)
    at /home/felipe/.ghcjs/x86_64-linux-0.1.0-7.8.3/ghcjs/thrunner.js:95:9
    at h$processQueue (/home/felipe/.ghcjs/x86_64-linux-0.1.0-7.8.3/ghcjs/thrunner.js:71:13)
    at Socket.<anonymous> (/home/felipe/.ghcjs/x86_64-linux-0.1.0-7.8.3/ghcjs/thrunner.js:51:17)
    at Socket.emit (events.js:117:20)
    at emitReadable_ (_stream_readable.js:427:10)
    at emitReadable (_stream_readable.js:423:5)
    at readableAddChunk (_stream_readable.js:166:9)

ReactJS.hs:27:12:
    Exception when trying to run compile-time code:
      Data.Binary.Get.runGet at position 0: demandInput: not enough bytes
    Code: Language.Haskell.TH.Quote.quoteExp
            jsx
            "\n\
            \        <div>Hello</div>\n\
            \        "
~~~

I tried other variants of the code and all fails with the same error.

**Comment by luite** [comment]
hmm, the foreign code for the `process` package should've been linked. I'll look into this.

**Comment by luite** [comment]
it seems to work here, using the ghcjsi branch, i might've fixed it at some point already

**Comment by fgaray** [comment]
I can't compile that branch. It gives me this error:

~~~

src/Compiler/DriverPipeline.hs:199:24:
    Not in scope: ‘isHsBootOrSig’

src/Compiler/DriverPipeline.hs:231:34:
    Not in scope: data constructor ‘Opt_WriteInterface’

src/Compiler/DriverPipeline.hs:233:41:
    Not in scope: ‘isHsBootOrSig’

src/Compiler/DriverPipeline.hs:250:20:
    Not in scope: data constructor ‘HsigFile’

src/Compiler/DriverPipeline.hs:423:53:
    Not in scope: type constructor or class ‘PackageKey’
    Perhaps you meant ‘PackageId’ (imported from Module)

src/Compiler/DriverPipeline.hs:459:31:
    Not in scope: type constructor or class ‘PackageKey’
    Perhaps you meant ‘PackageId’ (imported from Module)

src/Compiler/DriverPipeline.hs:982:41:
    ‘ms_iface_date’ is not a (visible) field of constructor ‘ModSummary’

src/Compiler/DriverPipeline.hs:1016:13:
    Not in scope: data constructor ‘HscUpdateSig’

src/Compiler/DriverPipeline.hs:1161:53:
    Not in scope: ‘basePackageKey’
    Perhaps you meant ‘basePackageId’ (imported from Module)

src/Compiler/DriverPipeline.hs:1624:49:
    Not in scope: ‘rtsPackageKey’
    Perhaps you meant ‘rtsPackageId’ (imported from Module)

src/Compiler/DriverPipeline.hs:1673:44:
    Not in scope: type constructor or class ‘PackageKey’
    Perhaps you meant ‘PackageId’ (imported from Module)

src/Compiler/DriverPipeline.hs:1714:29:
    Not in scope: type constructor or class ‘PackageKey’
    Perhaps you meant ‘PackageId’ (imported from Module)

src/Compiler/DriverPipeline.hs:1792:38:
    Not in scope: type constructor or class ‘PackageKey’
    Perhaps you meant ‘PackageId’ (imported from Module)

src/Compiler/DriverPipeline.hs:1798:23:
    Not in scope: ‘stringToPackageKey’
    Perhaps you meant ‘stringToPackageId’ (imported from Module)

src/Compiler/DriverPipeline.hs:1815:42:
    Not in scope: type constructor or class ‘PackageKey’
    Perhaps you meant ‘PackageId’ (imported from Module)

src/Compiler/DriverPipeline.hs:1818:51:
    Not in scope: type constructor or class ‘PackageKey’
    Perhaps you meant ‘PackageId’ (imported from Module)

src/Compiler/DriverPipeline.hs:1977:32:
    Not in scope: data constructor ‘ArchARM64’
    Perhaps you meant ‘ArchARM’ (imported from Platform)

src/Compiler/DriverPipeline.hs:2093:45:
    Not in scope: type constructor or class ‘PackageKey’
    Perhaps you meant ‘PackageId’ (imported from Module)

src/Compiler/DriverPipeline.hs:2103:48:
    Not in scope: type constructor or class ‘PackageKey’
    Perhaps you meant ‘PackageId’ (imported from Module)

src/Compiler/DriverPipeline.hs:2258:7:
    Not in scope: data constructor ‘Opt_WriteInterface’

src/Compiler/DriverPipeline.hs:2264:23:
    Not in scope: data constructor ‘HsigFile’

src/Compiler/DriverPipeline.hs:2288:41:
    Not in scope: ‘rtsPackageKey’
    Perhaps you meant ‘rtsPackageId’ (imported from Module)
cabal: Error: some packages failed to install:
ghcjs-0.1.0 failed during the building phase. The exception was:
ExitFailure 1
~~~

With this commands inside a sandbox:

~~~
$ git clone https://github.com/ghcjs/ghcjs-prim.git
$ git clone https://github.com/ghcjs/ghcjs.git
$ cabal install ./ghcjs ./ghcjs-prim
~~~

Using:
* ghc 7.8.4
* cabal 1.22.6.0

**Comment by luite** [comment]
You need GHC 7.10 for improved-base, apparently I forgot to update the lower bound. I'll do that with the next `ghcjs.cabal` file update.

Note that 7.10.1 has some problems with GHCJS and requires some workarounds. 7.10.2 fixes these problems.

**Comment by luite** [comment]
`improved-base` has been merged now, bumping the GHC requirement to 7.10.2, and I think it's probably fixed there. Can you open a new issue if you still see the problem?
